cirq/linalg/diagonalize.py

```diff
@@ -183,9 +183,9 @@ def bidiagonalize_real_matrix_pair_with_symmetric_products(
             raise ValueError('mat1 must be real.')
         if np.any(np.imag(mat2) != 0):
             raise ValueError('mat2 must be real.')
-        if not predicates.is_hermitian(mat1.dot(mat2.T), rtol=rtol, atol=atol):
+        if not predicates.is_hermitian(cast(np.ndarray, mat1.dot(mat2.T)), rtol=rtol, atol=atol):
             raise ValueError('mat1 @ mat2.T must be symmetric.')
-        if not predicates.is_hermitian(mat1.T.dot(mat2), rtol=rtol, atol=atol):
+        if not predicates.is_hermitian(cast(np.ndarray, mat1.T.dot(mat2)), rtol=rtol, atol=atol):
```

Comment:
Optional: These are the last two casts remaining. It'd be great if we could get rid of them, too.

Reply:
got rid of 'em!

We'll have to keep an eye out if the numpy folks change the signature for `np.dot` vs. `ndarray.dot`. It strikes me as odd that they're different.